### PR TITLE
refactor: ORM Builder pattern that supports select statement

### PR DESCRIPTION
**Pull request by smx-Morgan**

### CODE REVIEW

File: client/orm/qb/select.go [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+		cache: cache,
+	}
+}
+func (s *Selector[T]) Build() (*Query, error) {
+	var (
+		t   T
+		err error
+	)
+
+	s.model, _ = s.cache.GetByMd(&t)
```
Comment (flycash) [inline client/orm/qb/select.go:54]: Do not ignore this error

File: client/orm/qb/select.go [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+			typ := reflect.TypeOf(t)
+			s.sb.WriteByte('`')
+			s.sb.WriteString(typ.Name())
+			s.sb.WriteByte('`')
+		} else {
+			s.sb.WriteByte('`')
+			s.sb.WriteString(s.model.Table)
+			s.sb.WriteByte('`')
+		}
+	}
```
Comment (flycash) [inline client/orm/qb/select.go:88]: refactor this part that extract it as a new function

File: client/orm/qb/select.go [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+}
+func (s *Selector[T]) addArgs(args ...any) {
+	if s.args == nil {
+		s.args = make([]any, 0, 8)
+	}
+	s.args = append(s.args, args...)
+}
+
+func (s *Selector[T]) buildExpression(e Expression) error {
+	if e == nil {
```
Comment (flycash) [inline client/orm/qb/select.go:134]: you can add one more case:
```go
case nil:
    return nil
```

File: client/orm/qb/select.go [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+		}
+		if rp {
+			s.sb.WriteByte(')')
+		}
+	default:
+		return errs.NewErrUnsupportedExpressionType(exp)
+	}
+	return nil
+}
+func (s *Selector[T]) buildColumn(c Column, useAlias bool) error {
```
Comment (flycash) [inline client/orm/qb/select.go:175]: remove useAlias parameter, currently it always be true

File: client/orm/qb/select.go [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+	"github.com/beego/beego/v2/client/orm/internal/models"
+	"github.com/beego/beego/v2/client/orm/qb/errs"
+
+	"reflect"
+	"strings"
+)
+
+// The Selector is used to construct a SELECT statement
+type Selector[T any] struct {
+	sb        strings.Builder
```
Comment (flycash) [inline client/orm/qb/select.go:29]: using `internal/buffers`.

**Review by flycash — CHANGES_REQUESTED** [review]

File: client/orm/qb/query.go [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+)
+
+type Querier[T any] interface {
+	Get(ctx context.Context) (*T, error)
+	// batch query
+	GetMulti(ctx context.Context) (*T, error)
+}
+
+// Executor is used for INSERT , DELETE , UPDATE
+type Executor interface {
```
Comment (flycash) [inline client/orm/qb/query.go:29]: remove this interface because it's useless now

**Comment by flycash** [comment]
@smx-Morgan please check those comments.

**Comment by smx-Morgan** [comment]
got it, the code is currently being modified

**Comment by flycash** [comment]
and please resolve the conflicts

**Pull request closed without merging**